src/app/watchlist/handler/getWatchlistsHandler.go

```diff
@@ -0,0 +1,49 @@
+package handler
+
+import (
+	"net/http"
+
+	genericConstants "stock_broker_application/src/constants"
+	"watchlist/business"
+
+	"github.com/gin-gonic/gin"
+)
+
+type NewGetWatchlistController interface {
+	HandleGetWatchlists(ctx *gin.Context)
+}
+
+type getWatchlistController struct {
+	service business.NewGetWatchlistsService
+}
+
+func NewGetWatchListsController(service business.NewGetWatchlistsService) NewGetWatchlistController {
+	return &getWatchlistController{
+		service: service,
+	}
+}
+
+// @Summary Get the list of WatchLists
+// @Description Handler function to fetch the user's watchlist data.
+// @Produce json
+// @Success 200 {object} map[string]interface{} "Returns the user's watchlist data"
+// @Failure 500 {object} map[string]string "Internal server error"
+// @Security JWT
+// @Router /v1/watchlist/list [get]
+func (controller *getWatchlistController) HandleGetWatchlists(context *gin.Context) {
+
+	watchlistData, err := controller.service.NewGetWatchlistsService(context)
+
+	if err != nil {
+		context.JSON(http.StatusInternalServerError, gin.H{genericConstants.GenericJSONErrorMessage: err.Error()})
+		return
+	}
+
+	if len(watchlistData) == 0 {
+		context.JSON(http.StatusNoContent, gin.H{})
+		return
+	}
+
+	context.JSON(http.StatusOK, gin.H{genericConstants.WatchlistTable: watchlistData})
```

Comment:
```suggestion
	context.JSON(http.StatusOK, watchlistData})
```

Can you please try this